.github/workflows/clippy.yaml

```diff
@@ -0,0 +1,55 @@
+name: Release Static Checks
+
+on:
+  pull_request:
+    branches:
+      - msft-main
+
+jobs:
+  clippy:
+    name: Run Clippy on Rust Components
```

Comment:
The upstream static checks (static-checks.yaml) already run clippy - we should leverage those?